### PR TITLE
[DEM-1189] Improve how external adapters are added to factory

### PR DESCRIPTION
**Pull request by qSaevar**
* Added a method that allows single instrument adapters to be added to instrument adapter facotry
* Raise Type error if adapter to be added is not a subclass of InstrumentAdapter
* Change the name of method that allows import of whole adapter packages
* Only add classes that are subclasses of InstrumentAdapter

### CODE REVIEW

File: src/qilib/configuration_helper/instrument_adapter_factory.py [hunk 1]
```diff
@@ -32,18 +32,33 @@ class InstrumentAdapterFactory:
     instrument as well as a dictionary that provides key-value pairs of address specifications.
     """
 
-    instrument_adapters: Dict[Tuple[str, str], InstrumentAdapter] = {}
-    _external_adapters: Dict[str, ModuleType] = {}
+    adapter_instances: Dict[Tuple[str, str], InstrumentAdapter] = {}
```
Comment (peendebak) [inline src/qilib/configuration_helper/instrument_adapter_factory.py]: The `adapter_instances` is public, but not documented. I prefer docstrings associated to methods or attributes, but it seems there is no concensus on how to do that for class attributes. So I suggest documenting it in the class as
```
class InstrumentAdapterFactory:
"""
   xxx xxx 

   Attributes:
     adapter_instances: xxxx xxx
"""

File: src/tests/unittests/configuration_helper/test_instrument_adapter_factory.py [hunk 2]
```diff
@@ -31,20 +32,17 @@ def test_raise_value_error(self):
         with self.assertRaises(ValueError):
             InstrumentAdapterFactory.get_instrument_adapter('SomeAdapter', 'dev42')
 
-    def test_external_adapters_add_not_called(self):
-        from importlib import reload
-        from qilib.configuration_helper import instrument_adapter_factory
+    def test_adda_wrong_class_raises_type_error(self):
```
Comment (peendebak) [inline src/tests/unittests/configuration_helper/test_instrument_adapter_factory.py]: Typo?

File: src/qilib/configuration_helper/instrument_adapter_factory.py (hunk 1 above)
Comment (qSaevar) [reply src/qilib/configuration_helper/instrument_adapter_factory.py]: Maybe I should just go through the trouble of making this attribute private. There is no need to keep it public except for some unittests, but I'm sure I can refactor them.

File: src/tests/unittests/configuration_helper/test_instrument_adapter_factory.py (hunk 2 above)
Comment (qSaevar) [reply src/tests/unittests/configuration_helper/test_instrument_adapter_factory.py]: yes

**Pull request merged**